pod/annotations.go

```diff
@@ -159,6 +159,14 @@ const (
 	// service configuration
 
 	AnnotationKeyServicePrefix = "service.netflix.com"
+
+	// sidecar configuration
+
+	AnnotationKeyPrefixSidecars                      = "titusParameter.agent.sidecars"
+	AnnotationKeySuffixSidecarsChannelOverride       = "channel-override"
+	AnnotationKeySuffixSidecarsChannelOverrideReason = "channel-override-reason"
+	AnnotationKeySidecarsInclude                     = AnnotationKeyPrefixSidecars + "/include"
+	AnnotationKeyMutatorError                        = "mutatorErrors"
```

Comment:
Whoa, are we adding errors to a pod as an annotation instead of failing to create the pod?

How about `AnnotationKeyMutatorNonFatalErrors` then?

Reply:
I think the context for this was that the pod already exists, so we wanted to surface errors in some way that they were actually discoverable, instead of leaving the pod in a weird halfway state without context about why it didn't reach the final state.

But I don't think we've been very consistent about this - some types of errors are returned as fatal errors from the webhook instead of adding a `mutatorErrors` annotation to the pod. And as we discussed earlier today, these are now being surfaced in the UI.

So I think maybe we can remove this annotation and just return errors from the webhook. @slushie, does that make sense to you?

Reply:
:+1: Let's remove that annotation now that we can properly bubble errors up to users. We also have plenty of other places to track mutation errors (ie, Atlas and the output log) so I think our bases are covered.

Reply:
Great, removed the annotation from this PR (and will update the platform sidecar mutator not to use it when I update it to use the annotation key constants defined here).

Reply:
Yea if there is any fatal error, this pod will not last long, and nobody will see this annotation to inspect it.